SNIPS/snip-15.md

```diff
@@ -0,0 +1,40 @@
+# Draft---
+snip: 15
+title: Only an account can call its `execute`
+author: Ilia Volokh <iliav@starkware.co>
+discussions-to: 
+status: Draft
+type: Core
+created: 2024-01-08
+---
+
+## Simple Summary
+Restrict Starknet so only an account can call its own `execute` function, preventing internal calls from other contracts.
+
+## Abstract
+
+This proposal defines a protocol-level restriction that will be imposed on callers to the `execute` entrypoint of an account. The restriction requires the caller address to equal the address of the account of the `execute` in question.
+
+## Motivation
+
+Users act through their accounts via the `execute` entrypoint. Being able to access the execute of someone's account is tantamount to acting on their behalf. The default behavior should defend naive users and developers against such vulnerabilities.
+
+On the other hand, the current status quo is that all safe account implementations explicitly contain a line in their `execute` entrypoint which enforces that the call is not internal. Accounts without this line are exposed to having others act on their behalf.
+
+## Specification
+
+Execution clients will track the caller address and fail a transaction as soon as there is a call to an `execute` entrypoint whose caller address is not the associated account.
```

Comment:
Currently the caller is address zero right, if validate succeeds?

Reply:
True !